homework_4/numpy_challenge.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+
+
+if __name__ == "__main__":
+    """ here I created 4 np.arrays using 4 different methods just for training"""
+
+    fav_list = [196, 295, 394, 493]
+    fav_arr_1 = np.array(fav_list) # converted list to array
+
+    fav_arr_2 = np.zeros(5) # array with zeros
+
+    fav_arr_3 = np.arange(36, 46, 4) # like range, 4 is a step
+
+    fav_arr_4 = np.linspace(1, 10, 5) # like range, but 5 is the number of values
+    
+
+def matrix_multiplication(mtr_1, mtr_2):
+    """for two matrices multiplication if they can me multiplied in the original order"""
+
+    if np.shape(mtr_1)[1] == np.shape(mtr_2)[0]: # check that matrices can be multiplied
+        return np.matmul(mtr_1, mtr_2)
+
+    else:
+        return "ERROR: you have input two matrixes that can not be multiplied. Please check their "
+
+
+def multiplication_check(matrices):
+    """to check whether matrices from a list can be multiplied in the original order, output True or False"""
+
+    res = True # flag that we hope would not change :)
+
+    prev = np.shape(matrices[0]) # shape of the previous matrix
+    for i in range(1, len(matrices)):
+        curr = np.shape(matrices[i]) # shape of the current matrix
+        if prev[1] != curr[0]:
+            res = False # oh no flag changed, operation non valid
+            break
+        else:
+            prev = (prev[0], curr[1]) # multiplied matrix shape
+
+    return res
```

Comment:
Работает, но есть много лишних условий и переменных, можно сильно всё упростить
```suggestion
    for i in range(len(matrices) - 1):
        if matrices[i].shape[1] != matrices[i + 1].shape[0]:
            return False
    return True
```